Refactor: Refine Dashboard Cards - 764

hamed-valiollahi opened the pull request:
This PR refactors the dashboard cards to improve performance and maintainability by using a PostgreSQL materialized view, creating a unified API, moving transaction texts to locale files, and passing frontend filters in the navigate state.

@AlexZorkin I have a question about the Director Review card. Currently, there is no `Reviewed` status for Initiative Agreements. Are we planning to add this status in the future? Also, Please let me know If there are any other refinements.

Closes #764

AlexZorkin reviewed (approved):
This looks great ty. The dashboard will load much faster once this is merged. There won't be a Reviewed status for initiative agreements.

hamed-valiollahi commented:
> This looks great ty. The dashboard will load much faster once this is merged. There won't be a Reviewed status for initiative agreements.

Thank you, Alex, for reviewing my PR. I've made the following changes:
1. Fixed the migration order issue.
2. Updated the status for initiative agreements from "Reviewed" to "Approved".
3. Fixed a minor issue with the API routes.
